### DISCUSSION TITLE
Best way to run code before quasar dev?

### DISCUSSION THREAD

**Discussion by ssuess** [quasarframework/quasar#13353] [General - Components / Directives / etc] [answered]
I have a node script I run before launching quasar that compiles my i18n lang files. Ideally I would like to run this automatically whenever starting up quasar dev. What is the best approach to do this? A boot file? Some addition to the cli command? Some other script that calls both outside of quasar? 

For now I have just added a script to my package.json that runs one command then the other, but for some reason it feels like it might be better to put this compile code elsewhere, inside my app.

**Comment by metalsadman** [comment, marked as answer]
try it in `beforeXxx` hooks https://quasar.dev/quasar-cli-webpack/quasar-config-js#property-build.

**Reply by ssuess** [reply]
hm, I guess I can't run an external file or execute some command here? ie:
```
build: {
      beforeDev({ quasarConf }) {
        node combine-i18n.js
      },
```

**Reply by ssuess** [reply]
I just copied my entire code block in there and that worked, but it would be great to be able to execute an outside file here to keep it clean and separated.

**Reply by MilosPaunovic** [reply]
You can't run external files like that, but you can export whatever from that file, import it in `quasar.conf.js` and execute in `beforeDev`.
